### PR TITLE
Av1

### PR DESCRIPTION
**Pull request by gibix**

### CODE REVIEW

File: examples/matroska_remux.rs [hunk 1]
```diff
@@ -7,22 +7,23 @@ use matroska::{demuxer::MKV_DESC, muxer::MkvMuxer};
 use std::{fs::File, io::Cursor, sync::Arc};
 
 fn main() {
-    pretty_env_logger::init();
```
Comment (lu-zero) [inline examples/matroska_remux.rs]: Might be a good idea to spin out this change and make the remuxer take 2 arguments.

File: .travis.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ rust:
 
 matrix:
   allow_failures:
-    - rust: stable
+    - rust: nightly
```
Comment (lu-zero) [inline .travis.yml]: why do we need nightly?

File: src/muxer.rs [hunk 3]
```diff
@@ -512,6 +513,21 @@ pub fn stream_to_track(s: &Stream) -> TrackEntry {
             t.video = Some(Video {
                 pixel_width: v.width as u64,
                 pixel_height: v.height as u64,
+                colour: Some(Colour {
```
Comment (lu-zero) [inline src/muxer.rs]: the color change should land already.

File: src/demuxer.rs [hunk 4]
```diff
@@ -357,9 +357,11 @@ pub fn track_to_stream(info: &Info, t: &TrackEntry) -> Stream {
         info.timecode_scale as i64
     };
 
+    debug! {"Codec private: {:?}", t.codec_private};
+
     Stream {
         id: t.track_uid as isize,
-        index: 0,
+        index: t.stream_index,
```
Comment (lu-zero) [inline src/demuxer.rs]: this fix should land immediately.

File: src/demuxer.rs [hunk 5]
```diff
@@ -16,7 +16,8 @@ use av_format::{
     stream::Stream,
 };
 use log::{debug, error, trace};
-use nom::{self, Err, IResult, Offset};
```
Comment (lu-zero) [inline src/demuxer.rs]: format and cleanups should land immediately as well.

File: .travis.yml (hunk 2 above)
Comment (gibix) [reply .travis.yml]: for test, can fail

File: src/demuxer.rs (hunk 4 above)
Comment (gibix) [reply src/demuxer.rs]: also 469bd59  should be cherry picked

File: src/demuxer.rs [hunk 6]
```diff
@@ -178,12 +178,53 @@ fn track_entry_codec_id(t: &TrackEntry) -> Option<String> {
...
+                    false,
+                    0,
+                    8,
+                    0,
+                    1,
+                ),
+            ];
+
+            let f= Arc::new(Formaton::new(
+                ColorModel::CMYK, // where to check for other color scheme?
```
Comment (lu-zero) [inline src/demuxer.rs]: The color model is wrong. and it is codec-dependent sadly.

**Comment by Luni-4** [comment]
Closed in favor of #35

**Pull request closed without merging**